### PR TITLE
'js-error' event added

### PR DESCRIPTION
**Pull request by antpaw**
Now it's possible to trace js exceptions. It solves this https://github.com/segmentio/nightmare/issues/248 issue and makes Nightmare viable for TTD again.

### CODE REVIEW

File: lib/preload.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
 window.__nightmare = {};
 __nightmare.ipc = require('ipc');
 __nightmare.sliced = require('sliced');
+
+window.onerror = function (errorMsg, url, lineNumber, columnNumber, error) {
```
Comment (matthewmueller) [inline lib/preload.js]: probably want to do `window.addEventListener('error', fn)`

otherwise the page may overwrite this.

**Comment by matthewmueller** [comment]
So this would be for page errors, right? Because we already have: https://github.com/segmentio/nightmare/blob/master/lib/javascript.js#L25

**Comment by antpaw** [comment]
I've changed to addEventListener. i think the keyword 'js' is fine since its used here too `.inject('js', file)`

**Comment by antpaw** [comment]
I saw the callbacks for runtime errors https://github.com/segmentio/nightmare/blob/master/lib/runner.js#L106 but it's only used for `_inject` as far as i understand.
Right now i'm having a very strange issue and can not check if this two callbacks can be used as one.
The problem is, all the tests are passing within 400ms even if i have something like this `  title.should.equal('NOPE')` and i have 30 instances of Electron open in my doc. Even if i freshly checkout master form your remote, mocha tests are just behaving crazy. Do you have any idea why this could happen?

**Comment by antpaw** [comment]
Oh it looks like this issue is happing everywhere 
`$ make test` in this old build took 29s https://circleci.com/gh/segmentio/nightmare/364
`$ make test` in this build took 1s https://circleci.com/gh/segmentio/nightmare/373
since build 365 tests don't really run
what could have caused this? the commit for build 365 https://github.com/segmentio/nightmare/commit/58ff6acd00838576c50f953473101ec9241cb60f has nothing at all

**Comment by antpaw** [comment]
ok i've fixed this problem here https://github.com/segmentio/nightmare/pull/299

**Comment by antpaw** [comment]
Sorry i need to move this pr to a new branch https://github.com/segmentio/nightmare/pull/300

**Comment by mrself** [comment]
Looks like this does not work now. I manually created a page with a non-valid script and I can not track it with nitghtmare.

**Pull request closed without merging**